### PR TITLE
feat: Use jest-editor-support to show messages etc

### PR DESCRIPTION
**Pull request by orta**
This is going to be a WIP PR, although Danger isn't on this repo, so that might make it hard for me to preview 🗡

### CODE REVIEW

**Comment by orta** [comment]
This fixes an issue with TSC: https://github.com/facebook/jest/pull/4636

**Comment by orta** [comment]
Found a way to simplify even further

File: tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "declaration": true,
     "declarationDir": "dist",
-    "lib": ["es5", "es2015"],
+    "lib": ["es5", "es2015", "es2017"],
```
Comment (orta) [inline tsconfig.json]: Gives array more functions

File: src/index.ts [hunk 2]
```diff
@@ -1,36 +1,82 @@
...
+      // results.testResults.forEach((result) => {
+      // const testResults = results.testResults.filter((tr) => tr.testResults)
+      // reconciler.updateFileWithJestStatus({ testResults })
+      // // })
+      // const fails = reconciler.failedStatuses()
+      // console.log(reconciler)
+
+      // console.log(reconciler)
+
+      // fails.forEach((failingFile) => fail(fileToFailString(failingFile)))
```
Comment (orta) [inline src/index.ts]: struggling to get the data the same as vscode-jest - which is surprising, they both do the exact same thing ( run jest, output json, send it into the reconciler)

File: src/TestFailureFormatter.ts [hunk 3]
```diff
@@ -1,13 +0,0 @@
...
-export class TestFailureFormatter {
-  constructor(private results: any) {}
-
-  public format(): string {
-    return this.results.testResults
-      .filter((r: any) => r.failureMessage)
-      .map((r: any) => r.failureMessage)
-      .join('\n')
-  }
-}
```
Comment (orta) [inline src/TestFailureFormatter.ts]: There's a version of this inside jest-editor-support that does more work

**Comment by orta** [comment]
OK, this should be good as a rough first draft 👍 

Without seeing how Danger JS deals with it, I can't really progress much further.

File: src/types.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
+
```
Comment (orta) [inline src/types.ts]: Used JSON2TS to stub out the JSON shape

**Comment by macklinu** [comment]
At first glance, this looks awesome. Appreciate the contribution!

> Without seeing how Danger JS deals with it, I can't really progress much further.

I can set up Danger on this repo in another PR/commit and then we can merge it in here to try it out.

**Comment by orta** [comment]
Yeah, good call 👍

**Comment by macklinu** [comment]
#24 has been resolved and merged, so PRs should run Danger now. I tried to check out this branch locally and push a merge commit to your forked `master` branch, but I'm having trouble figuring out the proper way to do it (or if it's possible?).

**Comment by orta** [comment]
In theory you can, I also don't know how to do it - which is why I do branches, updating now

**Comment by orta** [comment]
interesting, looks like different versions of Jest test output has different keys available

**Comment by orta** [comment]
I'm going to leave this alone for a bit to think, but TBH, I think there's a good argument for letting these tests fail in master also ( so that there's always a visual of what it looks like )

**Comment by mxstbr** [comment]
I like that output :100:

**Comment by orta** [comment]
OK, I'm pretty convinced on this now, 👍 

I've skipped the tests which fail.

**Comment by macklinu** [comment]
:shipit:

**Comment by mxstbr** [comment]
New release?! 💯

**Comment by macklinu** [comment]
Yep! Give 1.1.0 a try and leave any feedback/bug reports in a new issue. ❤️

**Pull request merged**